### ISSUE TITLE
MacOS code signing fails with WebRTC.framework: bundle format is ambiguous (could be app or framework)

### ISSUE THREAD

**Issue by GigaDroid** [flutter-webrtc/flutter-webrtc#421]
Hey, I'm unable to build my application for MacOS due to a code signing problem related to WebRTC.framework

This is the build output:

```
Running "flutter pub get" in flair...
Launching lib/main.dart on macOS in debug mode...
Building macOS application...
mkdir -p /Users/leonkukuk/Documents/Projects/flair/flair/build/macos/Build/Products/Debug/flair.app/Contents/Frameworks
rsync --delete -av --filter P .*.?????? --links --filter "- CVS/" --filter "- .svn/" --filter "- .git/" --filter "- .hg/" --filter "- Headers" --filter "- PrivateHeaders" --filter "- Modules" "/Users/leonkukuk/Documents/Projects/flair/flair/macos/Pods/../Flutter/ephemeral/.symlinks/flutter/darwin-x64/FlutterMacOS.framework" "/Users/leonkukuk/Documents/Projects/flair/flair/build/macos/Build/Products/Debug/flair.app/Contents/Frameworks"
building file list ... done
FlutterMacOS.framework/
FlutterMacOS.framework/Versions/
FlutterMacOS.framework/Versions/A/
FlutterMacOS.framework/Versions/A/FlutterMacOS
FlutterMacOS.framework/Versions/A/Resources/
FlutterMacOS.framework/Versions/A/Resources/Info.plist
FlutterMacOS.framework/Versions/A/Resources/icudtl.dat
FlutterMacOS.framework/Versions/A/_CodeSignature/
FlutterMacOS.framework/Versions/A/_CodeSignature/CodeResources

sent 39643117 bytes  received 138 bytes  79286510.00 bytes/sec
total size is 39637671  speedup is 1.00
Destination binary is symlinked...
Code Signing /Users/leonkukuk/Documents/Projects/flair/flair/build/macos/Build/Products/Debug/flair.app/Contents/Frameworks/FlutterMacOS.framework with Identity -
/usr/bin/codesign --force --sign -  --preserve-metadata=identifier,entitlements '/Users/leonkukuk/Documents/Projects/flair/flair/build/macos/Build/Products/Debug/flair.app/Contents/Frameworks/FlutterMacOS.framework'
/Users/leonkukuk/Documents/Projects/flair/flair/build/macos/Build/Products/Debug/flair.app/Contents/Frameworks/FlutterMacOS.framework: replacing existing signature
rsync --delete -av --filter P .*.?????? --links --filter "- CVS/" --filter "- .svn/" --filter "- .git/" --filter "- .hg/" --filter "- Headers" --filter "- PrivateHeaders" --filter "- Modules" "/Users/leonkukuk/Documents/Projects/flair/flair/macos/Pods/../Flutter/ephemeral/.symlinks/plugins/flutter_webrtc/macos/WebRTC.framework" "/Users/leonkukuk/Documents/Projects/flair/flair/build/macos/Build/Products/Debug/flair.app/Contents/Frameworks"
building file list ... done

sent 376 bytes  received 20 bytes  792.00 bytes/sec
total size is 44033619  speedup is 111196.01
Code Signing /Users/leonkukuk/Documents/Projects/flair/flair/build/macos/Build/Products/Debug/flair.app/Contents/Frameworks/WebRTC.framework with Identity -
/usr/bin/codesign --force --sign -  --preserve-metadata=identifier,entitlements '/Users/leonkukuk/Documents/Projects/flair/flair/build/macos/Build/Products/Debug/flair.app/Contents/Frameworks/WebRTC.framework'
/Users/leonkukuk/Documents/Projects/flair/flair/build/macos/Build/Products/Debug/flair.app/Contents/Frameworks/WebRTC.framework: bundle format is ambiguous (could be app or framework)
Command PhaseScriptExecution failed with a nonzero exit code
note: Using new build system
note: Building targets in parallel
note: Planning build
note: Constructing build description
** BUILD FAILED **

Exception: Build process failed
```

* **Flutter version**: 
`[✓] Flutter (Channel dev, 1.24.0-10.2.pre, on macOS 11.0.1 20B29 darwin-x64, locale en-DE)
    • Flutter version 1.24.0-10.2.pre at /Users/Shared/Flutter/flutter
    • Framework revision 022b333a08 (10 days ago), 2020-11-18 11:35:09 -0800
    • Engine revision 07c1eed46b
    • Dart version 2.12.0 (build 2.12.0-29.10.beta)
`
* **Plugin version**: 0.5.7
* **OS**: MacOS
* **OS version**: 11.0.1

**Comment by jascodes** [comment]
same

**Comment by cloudwebrtc** [comment]
@jascodes @GigaDroid Pub does not support soft links, so `macos/WebRTC.framework/` in the plug-in downloaded from pub will generate a directory instead of a soft link, so you need to manually delete the `macos/WebRTC.framework/Versions` directory to compile.

**Comment by GigaDroid** [comment]
<img width="496" alt="Screenshot 2020-11-29 at 01 32 34" src="https://user-images.githubusercontent.com/1688752/100528953-cc023a00-31e2-11eb-9091-c0ed0497f247.png">
This is the directory you are talking about, right? I deleted it and ran codesign again and I wasn't able to start the application. This is the MacOS Problem Report:
```
Process:               flair [57912]
Path:                  /Users/USER/Documents/*/flair.app/Contents/MacOS/flair
Identifier:            com.example.flair
Version:               1.0.0 (1)
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           flair [57912]
User ID:               501

Date/Time:             2020-11-29 01:18:55.433 +0100
OS Version:            macOS 11.0.1 (20B29)
Report Version:        12
Anonymous UUID:        4ADBEC08-3DC1-F0B1-7B67-0807169EAAD4

Sleep/Wake UUID:       8C22B383-0313-44F2-BAC5-4E4ACA9F8BA8

Time Awake Since Boot: 63000 seconds
Time Since Wake:       27000 seconds

System Integrity Protection: enabled

Crashed Thread:        0

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    DYLD, [0x1] Library missing

Application Specific Information:
dyld: launch, loading dependent libraries

Dyld Error Message:
  dyld: Using shared cache: 37FEA16C-BED8-3B52-974D-F87051F072C8
Library not loaded: @rpath/flutter_webrtc.framework/Versions/A/flutter_webrtc
  Referenced from: /Users/USER/Documents/*/flair.app/Contents/MacOS/flair
  Reason: image not found

Binary Images:
       0x100e1e000 -        0x100e21fff +com.example.flair (1.0.0 - 1) <DF28CABB-0F99-32DB-94F5-14F0E0EB5889> /Users/USER/Documents/*/flair.app/Contents/MacOS/flair
       0x1014b2000 -        0x10154dfff  dyld (832.7.1) <2705F0D8-C104-3DE9-BEB5-B1EF6E28656D> /usr/lib/dyld
       0x1015ca000 -        0x103155fff +io.flutter.flutter-macos (1.0 - 1.0) <6666B5B2-3E85-3486-8B68-EECEAE27D7B4> /Users/USER/Documents/*/flair.app/Contents/Frameworks/FlutterMacOS.framework/Versions/A/FlutterMacOS
       0x103aca000 -        0x104321fff +org.webrtc.WebRTC (1.0 - 1.0) <29E68C2B-1F17-325D-81D0-6AA77D140AEB> /Users/USER/Documents/*/flair.app/Contents/Frameworks/WebRTC.framework/WebRTC

Model: MacBookPro11,1, BootROM 427.0.0.0.0, 2 processors, Dual-Core Intel Core i5, 2.6 GHz, 8 GB, SMC 2.6f59
Graphics: kHW_IntelHD4000Item, Intel HD Graphics 4000, spdisplays_builtin
Memory Module: BANK 0/DIMM0, 4 GB, DDR3, 1600 MHz, 0x02FE, 0x45424A3831554738454255302D474E2D4620
Memory Module: BANK 1/DIMM0, 4 GB, DDR3, 1600 MHz, 0x02FE, 0x45424A3831554738454255302D474E2D4620
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0x10F), Broadcom BCM43xx 1.0 (7.77.111.1 AirPortDriverBrcmNIC-1669)
Bluetooth: Version 8.0.1f5, 3 services, 27 devices, 1 incoming serial ports
Network Service: Wi-Fi, AirPort, en4
Serial ATA Device: APPLE SSD SM256E, 251 GB
USB Device: USB 2.0 Bus
USB Device: Hub
USB Device: FaceTime HD Camera (Built-in)
USB Device: USB 2.0 Bus
USB Device: Hub
USB Device: Hub
USB Device: Apple Internal Keyboard / Trackpad
USB Device: BRCM20702 Hub
USB Device: Bluetooth USB Host Controller
USB Device: USB 3.0 Bus
Thunderbolt Bus: MacBook Pro, Apple Inc., 23.4
```

**Comment by GigaDroid** [comment]
Nevermind the directory you were talking about was in `/usr/local/flutter/.pub-cache/hosted/pub.dartlang.org/`. It's working now! However, I think the issue can be left open as this is more of a workaround and there should be a proper fix I guess?

**Comment by cloudwebrtc** [comment]
This must refactor the `WebRTC.framework` directory, remove the  `Versions` directory, and replace the soft link with the real file/directory.
But this will cause another issue https://github.com/flutter-webrtc/flutter-webrtc/pull/299.
